omod/src/main/java/org/bahmni/module/hip/web/service/FHIRResourceMapper.java

```diff
@@ -41,8 +46,24 @@ public Practitioner mapToPractitioner(EncounterProvider encounterProvider) {
         return practitionerTranslatorProvider.toFhirResource(encounterProvider.getProvider());
     }
 
+    private String displayName(Object object) {
+        if (object == null)
+            return "";
+        return object.toString() + " ";
+
+    }
+
     public MedicationRequest mapToMedicationRequest(DrugOrder order) {
-        return medicationRequestTranslator.toFhirResource(order);
+        String dosingInstrutions = displayName(order.getDose()) +
+                displayName(order.getDoseUnits().getName()) +
```

Comment:
What if the getDoseUnits() return null? getName() on getDoseUnits() would lead to NullPointerException